src/transaction/TransactionController.ts

```diff
@@ -402,7 +432,7 @@ export class TransactionController extends BaseController<TransactionConfig, Tra
       networkID,
       chainId,
       origin,
-      status: 'unapproved',
+      status: TransactionStatus.unapproved as TransactionStatus.unapproved,
```

Comment:
I'm not sure why `as` was required here. If I omitted it, it gave me a type error that claimed it wasn't a valid status :confused:

Reply:
that's very strange, but if it works it works!

Reply:
Well, to be more specific: it fails because the inferred type is `TransactionStatus`, not `TransactionStatus.unapproved`. So it won't let me pass this transaction meta object into `failTransaction` because it's missing an `error` property, which might be required if it has a `failed` status.

Saying `as TransactionStatus.unapproved` forces it to assign it a more specific type, which resolves the type error.